pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotRealtimeSegmentManager.java

```diff
@@ -191,6 +191,27 @@ private synchronized void assignRealtimeSegmentsToServerInstancesIfNecessary()
               RealtimeSegmentZKMetadata realtimeSegmentZKMetadata = ZKMetadataProvider
                   .getRealtimeSegmentZKMetadata(_pinotHelixResourceManager.getPropertyStore(), segName.getTableName(),
                       partition);
+              if (realtimeSegmentZKMetadata == null) {
+                // jfim: If we end up here, this means that for some reason we can't read the segment metadata from the
+                // property store. This could be because of a transient ZK error, because someone deleted the property
+                // store entry after we read the ideal state or because someone deleted the property store manually
+                // without updating the ideal state.
+                //
+                // In this case, we assume that the ideal state contains an appropriate state that reflects whether or
+                // not the segment was still in progress.
+                LOGGER.warn("Failed to read segment metadata for segment {}", partition);
+
+                // Since the ideal state update is atomic, we assume that if one partition is consuming, they're all
+                // consuming. As such, we don't need to reassign a new segment for this Kafka partition.
+                if (state.getInstanceStateMap(partition).values().contains("CONSUMING")) {
```

Comment:
the partition will not be in CONSUMING state because we consider only HLC segments in this code path